examples/ssh/opkssh/cli.go

```diff
@@ -0,0 +1,319 @@
+package main
+
+import (
+	"context"
+	"crypto"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/exp/slices"
+
+	"os"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/openpubkey/openpubkey/client"
+	"github.com/openpubkey/openpubkey/client/providers"
+	"github.com/openpubkey/openpubkey/examples/ssh/sshcert"
+	"github.com/openpubkey/openpubkey/pktoken"
+	"github.com/openpubkey/openpubkey/util"
+	"golang.org/x/crypto/ssh"
+)
+
+var (
+	key              = []byte("NotASecureKey123")
+	clientID         = "184968138938-g1fddl5tglo7mnlbdak8hbsqhhf79f32.apps.googleusercontent.com"
+	requiredAudience = "184968138938-g1fddl5tglo7mnlbdak8hbsqhhf79f32.apps.googleusercontent.com"
+	// The clientSecret was intentionally checked in for the purposes of this example,. It holds no power. Do not report as a security issue
+	clientSecret = "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" // Google requires a ClientSecret even if this a public OIDC App
+	issuer       = "https://accounts.google.com"
+	scopes       = []string{"openid profile email"}
+	redirURIPort = "3000"
+	callbackPath = "/login-callback"
+	redirectURI  = fmt.Sprintf("http://localhost:%v%v", redirURIPort, callbackPath)
+)
+
+// This code is currently intended as an example for how OpenPubkey can secure SSH access.
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Printf("Example SSH key generator using OpenPubkey: command choices are: login, ver")
+		return
+	}
+	command := os.Args[1]
+
+	op := providers.GoogleOp{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		Issuer:       issuer,
+		Scopes:       scopes,
+		RedirURIPort: redirURIPort,
+		CallbackPath: callbackPath,
+		RedirectURI:  redirectURI,
+	}
+
+	switch command {
+	case "login":
+		{
+			if len(os.Args) != 2 {
+				fmt.Println("Invalid number of arguments for login, should be `opkssh login`")
+				os.Exit(1)
+			}
+
+			// If principals is empty the server does not enforce any principal.
+			// The OPK verifier should use policy to make this decision.
+			principals := []string{}
+
+			gqFalse := false
+			alg := jwa.ES256
+
+			signer, err := util.GenKeyPair(alg)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+
+			client := &client.OpkClient{
+				Op: &op,
+			}
+
+			certBytes, seckeySshPem, err := CreateSSHCert(context.Background(), client, signer, alg, gqFalse, principals)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+
+			// Write ssh secret key and public key to filesystem
+			err = WriteKeysToSSHDir(seckeySshPem, certBytes)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+			os.Exit(0)
+		}
+	case "ver":
```

Comment:
why couldn't we make this "verify"? When I first saw this, by mind went to "version"

Reply:
SIGN, VER are the common function names for signing and verification in digital signatures. Shorter names in clis and commands are to be preferred: `'ls`, `cat`, `rm`. Why make someone type more and slow them down?

Reply:
If you want to argue the best interface for the opkssh that is fine, create a PR, but this is a simple example usecase not a tool. It doesn't have an arg parser. It doesn't have a real logger.